File: go/exercise01/saying_hello.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)

	fmt.Print(("What is your name? "))
	scanner.Scan()
	sayHello(scanner.Text())
}

func sayHello(name string) {
	fmt.Printf("Hello, %v, nice to meet you!\n", name)
}

File: zig/saying_hello.zig
const std = @import("std");

pub fn main() !void {
    const stdin = std.io.getStdIn().reader();
    var buffer: [20]u8 = undefined;

    std.debug.print("What is your name? ", .{});
    const name = try stdin.readUntilDelimiter(&buffer, '\n');

    std.debug.print("Hello, {s}, nice to meet you!", .{name});
}
